util/dvsim/FlowCfg.py

```diff
@@ -443,6 +447,35 @@ def write_results_html(self, filename, text_md):
         # Link the `/latest` folder with the latest reports.
         mk_symlink(result_path, symlink_path)
 
+    def write_results_hjson(self, test_hjson):
+        ''' Merge and write test_hjson to results.hjson.
+
+        First check if there are existing `results.hjson` file. If exists, append the current
+        results to the list of previous results hjson. Remove the oldest timestamps if
+        `results.hjson` file contains more than 120 entries.
+        '''
+        result_path = os.path.join(self.results_path, "results.hjson")
+        symlink_path = os.path.join(self.symlink_path, "results.hjson")
+
+        # If `results.hjson` exists, append the current hjson result and remove
+        # older entries if more than 120 entries are in the file.
+        if (os.path.isfile(symlink_path)):
+            move(symlink_path, result_path)
+            with open(result_path, 'rU') as f:
+                text = f.read()
+                hjson_results_dict = hjson.loads(text, use_decimal=True)
+                while (len(hjson_results_dict)) > 120:
+                    first_key = next(iter(hjson_results_dict))
+                    del hjson_results_dict[first_key]
+                hjson_results_dict[self.timestamp] = test_hjson[self.timestamp]
```

Comment:
What happens if there are two tests that happen to have the same timestamp? Maybe it would be better to serialize as a list of pairs `(timestamp, results)`?